src/cryptography/hazmat/backends/openssl/ec.py

```diff
@@ -302,3 +322,120 @@ def public_bytes(self, encoding, format):
             self._evp_pkey,
             None
         )
+
+    def __eq__(self, public_key):
+        sg = self._backend._lib.EC_KEY_get0_group(self._ec_key)
+        self._backend.openssl_assert(sg != self._backend._ffi.NULL)
+
+        sp = self._backend._lib.EC_KEY_get0_public_key(self._ec_key)
+        self._backend.openssl_assert(sp != self._backend._ffi.NULL)
+
+        og = self._backend._lib.EC_KEY_get0_group(public_key._ec_key)
+        self._backend.openssl_assert(og != self._backend._ffi.NULL)
+
+        op = self._backend._lib.EC_KEY_get0_public_key(public_key._ec_key)
+        self._backend.openssl_assert(op != self._backend._ffi.NULL)
+
+        with self._backend._tmp_bn_ctx() as bn_ctx:
+            res = self._backend._lib.EC_GROUP_cmp(sg, og, bn_ctx)
+            if res != 0:
+                return False
+
+            res = self._backend._lib.EC_POINT_cmp(sg, sp, op, bn_ctx)
+
+        return res == 0
+
+
+    def __neg__(self):
+        group = self._backend._lib.EC_KEY_get0_group(self._ec_key)
+        self._backend.openssl_assert(group != self._backend._ffi.NULL)
+
+        point = self._backend._lib.EC_KEY_get0_public_key(self._ec_key)
+        self._backend.openssl_assert(point != self._backend._ffi.NULL)
+
+        inv = self._backend._lib.EC_POINT_dup(point, group)
+        self._backend.openssl_assert(inv != self._backend._ffi.NULL)
+        inv = self._backend._ffi.gc(inv, self._backend._lib.EC_POINT_free)
+
+        with self._backend._tmp_bn_ctx() as bn_ctx:
+            res = self._backend._lib.EC_POINT_invert(group, inv, bn_ctx)
+            self._backend.openssl_assert(res == 1)
+
+        return _point_to_public_key(self._backend, group, inv)
+
+    def __add__(self, public_key):
+        sg = self._backend._lib.EC_KEY_get0_group(self._ec_key)
+        self._backend.openssl_assert(sg != self._backend._ffi.NULL)
+
+        sp = self._backend._lib.EC_KEY_get0_public_key(self._ec_key)
+        self._backend.openssl_assert(sp != self._backend._ffi.NULL)
+
+        og = self._backend._lib.EC_KEY_get0_group(public_key._ec_key)
+        self._backend.openssl_assert(og != self._backend._ffi.NULL)
+
+        op = self._backend._lib.EC_KEY_get0_public_key(public_key._ec_key)
+        self._backend.openssl_assert(op != self._backend._ffi.NULL)
+
+        sum = self._backend._lib.EC_POINT_new(sg)
+        self._backend.openssl_assert(sum != self._backend._ffi.NULL)
+        sum = self._backend._ffi.gc(sum, self._backend._lib.EC_POINT_free)
+
+        with self._backend._tmp_bn_ctx() as bn_ctx:
+            res = self._backend._lib.EC_GROUP_cmp(sg, og, bn_ctx)
+            self._backend.openssl_assert(res == 0)
+
+            res = self._backend._lib.EC_POINT_add(sg, sum, sp, op, bn_ctx)
+            self._backend.openssl_assert(res == 1)
+
+        return _point_to_public_key(self._backend, sg, sum)
+
+    def __sub__(self, public_key):
+        return self + -public_key
+
+    def __multiply(self, bignum):
```

Comment:
We typically use just a single underscore to denote a private method, but obviously `__` will make it genuinely inaccessible outside the class. I don't believe we do this anywhere else in cryptography but I'm okay with it. Others may disagree.